zstacklib/zstacklib/utils/gpu.py

```
@@ -593,6 +593,126 @@ def post_process_enflame_gpu_device(to):
 
+def get_kunlunxin_gpu_xpu_id_cmd():
+    return "npu-smi -L"
+
+
+def get_kunlunxin_xpu_id(xpu_id_output):
+    xpu_ids = []
+    for line in xpu_id_output.splitlines():
+        line = line.strip()
+        if not line:
+            continue
+        match = re.match(r'^XPU\s+(\d+):', line)
```

Ruff (0.14.6)

`re` may be undefined, or defined from star imports

(F405)